org.envirocar.app/src/org/envirocar/app/views/obdselection/OBDSelectionFragment.java

```diff
@@ -506,6 +508,9 @@ public void onPairingError(BluetoothDevice device) {
                         }
                         if (text != null)
                             text.setText(device.getName());
+
+                        // If error occured then, enable the click for the device.
+                        rediscover();
```

Comment:
No need to initiate a rediscover here. However, I see why you are doing so: since you are setting the _onClickListener_ to `null`, there is no easy way, to activate the click action again. Maybe, you can find an alternative approach for disabling and activating the click event on the view that represents the pairing device.

Reply:
I see that when we click no,in the pair device dialog, bluetooth discovery starts again. Hence, I followed similar approach, if not connected/error after pairing started then rediscover. 

https://user-images.githubusercontent.com/70392921/130388118-8da34923-c3ef-41b9-8e81-d17d3fbd0923.mp4